misk-action-scopes/src/main/kotlin/misk/scope/ActionScope.kt

```diff
@@ -97,15 +97,18 @@ class ActionScope @Inject internal constructor(
   }
 
   /** Creates a new scope on the current thread with the provided seed data */
-  fun create(seedData: Map<Key<*>, Any?>): Instance {
+  fun create(
+    seedData: Map<Key<*>, Any?>,
+    providerOverrides: Map<Key<*>, ActionScopedProvider<*>> = emptyMap(),
+  ): Instance {
     check(!inScope()) {
       "cannot create an ActionScope.Instance on a thread that is already running in an action scope"
     }
 
     val immediateValues = seedData.mapValues { (_, value) -> ImmediateLazy(value) }
 
     val lazyValues = providers.mapValues { (key, _) ->
-      SynchronizedLazy(providerFor(key))
+      SynchronizedLazy(providerFor(key, providerOverrides))
```

Comment:
Instead of passing the overrides, can we just update line 110 to be `(providers + providerOverrides).mapValues` instead?

Reply:
Good push. I added another block that contains `lazyOverrides` because `providers` contains `Provider<ActionScopedProvider<*>>` so the types won't mesh